packages/calling-stateful-client/src/StreamUtils.ts

```diff
@@ -472,21 +506,48 @@ function disposeViewUnparentedVideo(
   internalContext: InternalCallContext,
   stream: LocalVideoStreamState
 ): void {
+  _logEvent(callingStatefulLogger, {
+    name: EventNames.START_DISPOSE_LOCAL_STREAM,
+    level: 'info',
+    message: 'Start disposing unparented local stream.'
+  });
+
   context.deleteDeviceManagerUnparentedView(stream);
 
   const renderInfo = internalContext.getUnparentedRenderInfo(stream);
   if (!renderInfo) {
+    _logEvent(callingStatefulLogger, {
+      name: EventNames.LOCAL_DISPOSE_INFO_NOT_FOUND,
+      level: 'error',
+      message: 'Cannot find render info when disposing unparented local stream.'
+    });
     return;
   }
 
   if (renderInfo.status === 'Rendering') {
+    _logEvent(callingStatefulLogger, {
+      name: EventNames.LOCAL_STREAM_STOPPING,
+      level: 'warning',
+      message: 'Unparented local stream is still rendering. Changing status to stopping.'
+    });
     internalContext.setUnparentedRenderInfo(stream, renderInfo.stream, 'Stopping', undefined);
   } else {
     internalContext.deleteUnparentedRenderInfo(stream);
   }
 
   if (renderInfo.renderer) {
+    _logEvent(callingStatefulLogger, {
+      name: EventNames.DISPOSING_LOCAL_RENDERER,
+      level: 'info',
+      message: 'Disposing unparented local view renderer.'
+    });
     renderInfo.renderer.dispose();
+  } else {
+    _logEvent(callingStatefulLogger, {
+      name: EventNames.LOCAL_RENDERER_NOT_FOUND,
+      level: 'warning',
```

Comment:
Same comment as above.

It's unfortunate we have duplicate code-paths for such closely related non-trivial operations. It's an invitation to bugs...

Reply:
updated, yeah I agree on the redundancy but also can't commit time to making it better, the code is so similar we should be able to have a local | remote type and keep it somewhat agnostic